x-pack/plugins/enterprise_search/public/applications/app_search/components/source_engines/source_engines_logic.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { kea, MakeLogicType } from 'kea';
+
+import { Meta } from '../../../../../common/types';
+
+import { flashAPIErrors } from '../../../shared/flash_messages';
+import { HttpLogic } from '../../../shared/http';
+import { EngineLogic } from '../engine';
+import { EngineDetails } from '../engine/types';
+
+interface EnginesAPIResponse {
+  results: EngineDetails[];
+  meta: Meta;
+}
+
+interface SourceEnginesLogicValues {
+  dataLoading: boolean;
+  sourceEngines: EngineDetails[];
+}
+
+interface SourceEngineLogicActions {
+  fetchSourceEngines: () => void;
+  setSourceEngines: (
```

Comment:
[Optional, just throwing this out there] @JasonStoltz  is better than me at describing the benefits of declarative vs imperative action names in Kea/redux actions, but if we wanted to make this fn name more event-driven and less imperative it could be something like `onSourceEnginesFetch()`